File: backend/app/models/submission.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, DateTime, ForeignKey, Text, Integer
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database import Base


class Submission(Base):
    __tablename__ = "submissions"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    question_id = Column(UUID(as_uuid=True), ForeignKey("questions.id", ondelete="CASCADE"), nullable=False)
    child_name = Column(String, nullable=False)
    answer_given = Column(Text, nullable=False)
    score = Column(Integer, nullable=True)  # 0 to 100
    feedback = Column(Text, nullable=True)
    suggestions = Column(Text, nullable=True)
    submitted_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    # Relationships
    question = relationship("Question", back_populates="submissions")

File: frontend/src/types/submission.ts
export interface Submission {
  id: string;
  question_id: string;
  child_name: string;
  child_id?: string | null;
  difficulty_level?: string | null;
  answer_given: string;
  score: number | null;
  feedback: string | null;
  suggestions: string | null;
  submitted_at: string;
}

export interface SubmissionDetail extends Submission {
  question: string;
}
